skdownscale/pointwise_models/groupers.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import pandas as pd
+
+
+class SkdownscaleGroupGeneratorBase:
+    pass
+
+
+class PaddedDOYGrouper(SkdownscaleGroupGeneratorBase):
+    def __init__(self, df, offset=15):
+        self.df = df
+        self.offset = offset
+        self.max = 365
+        self.days_of_year = np.arange(1, 366)
```

Comment:
Which calendars will this support? I'm thinking it will be useful to check that the calendar of df.index is valid for this grouper method.